java-util/src/main/java/io/druid/java/util/common/io/NativeIO.java

```diff
@@ -149,21 +203,33 @@ public static void chunkedCopy(InputStream src, File dest) throws IOException
 
     final byte[] buf = new byte[8 << 20]; // 8Mb buffer
     long offset = 0;
+    long lastOffset = 0;
 
     try (
-      final RandomAccessFile raf = new RandomAccessFile(dest, "rwd")
+      final RandomAccessFile raf = new RandomAccessFile(dest, "rw")
     ) {
       final int fd = getfd(raf.getFD());
 
-      for (int numBytes = 0, bytesRead = 0; bytesRead > -1; ) {
+      for (int numBytes = 0, bytesRead = 0, lastBytes = 0; bytesRead > -1;) {
         bytesRead = src.read(buf, numBytes, buf.length - numBytes);
 
         if (numBytes >= buf.length || bytesRead == -1) {
           raf.write(buf, 0, numBytes);
-          trySkipCache(fd, offset, numBytes);
+
+          /* This won't block, but will start writeout asynchronously */
```

Comment:
I think we haven't formalized it, but `//` comments are much more typical in Druid sources.